util.js

```diff
@@ -41,6 +41,37 @@ function enum_from_settings(nick, enum_class) {
     return enum_class[nick.replace(/-/g, '_').toUpperCase()];
 }
 
+function show_dialog(parent_window, message, message_type = Gtk.MessageType.ERROR) {
+    const dialog = new Gtk.MessageDialog({
+        transient_for: parent_window,
+        modal: true,
+        buttons: Gtk.ButtonsType.CLOSE,
+        message_type,
+        text: message,
+    });
+    dialog.connect('response', () => dialog.destroy());
+    dialog.show();
+}
+
+// Looks up a `Gio.SettingsSchema` with the identifier `schema_id`.
+// Throw error if the schema does not exist.
+function get_settings_schema(schema_id) {
+    const schema_source = Gio.SettingsSchemaSource.get_default();
+    const schema = schema_source.lookup(schema_id, true);
+    if (schema === null)
+        throw new Error(`Settings schema '${schema_id}' not found.`);
+    return schema;
+}
+
+// Get key named `name` from `schema`.
+// Throw error if the key does not exist.
+function get_settings_schema_key(schema, name) {
+    if (!schema.has_key(name))
+        throw new Error(`Key '${name}' does not exist in settings schema '${schema.get_id()}'.`);
+
+    return schema.get_key(name);
+}
```

Comment:
Please keep all these functions in `prefs.js` - currently they aren't used by anything outside `prefs.js` (and probably won't ever be).